File: Easy/2026-06-01-2144-Minimum-Cost-of-Buying-Candies-With-Discount/solution.js
function minimumCost(cost) {
    cost.sort((a, b) => b - a);
    let totalCost = 0;

    for (let i = 0; i < cost.length; i++) {
        totalCost += cost[i];
        if ((i + 1) % 3 === 0) {
            totalCost -= cost[i]; // Remove the cost of the free candy
        }
    }

    return totalCost;
}

File: Easy/2026-06-01-2144-Minimum-Cost-of-Buying-Candies-With-Discount/solution.java
class Solution {
    public int minimumCost(int[] cost) {
        Arrays.sort(cost);
        int totalCost = 0;
        int n = cost.length;

        for (int i = n - 1; i >= 0; i--) {
            totalCost += cost[i];
            if (i - 2 >= 0) {
                i--; // Skip the next candy as it will be free
            }
        }

        return totalCost;
    }
}